packages/llamaindex/src/storage/vectorStore/UpstashVectorStore.ts

```diff
@@ -0,0 +1,203 @@
+import {
+  VectorStoreBase,
+  type IEmbedModel,
+  type MetadataFilters,
+  type VectorStoreNoEmbedModel,
+  type VectorStoreQuery,
+  type VectorStoreQueryResult,
+} from "./types.js";
+
+import type { BaseNode, Metadata, TextNode } from "@llamaindex/core/schema";
+import { getEnv } from "@llamaindex/env";
+import { Index } from "@upstash/vector";
+import { metadataDictToNode, nodeToMetadata } from "./utils.js";
+
+type UpstashParams = {
+  namespace?: string;
+  token: string;
```

Comment:
adding docs that it defaults to `process.env.UPSTASH_VECTOR_REST_TOKEN` and `process.env.UPSTASH_VECTOR_REST_URL` would be good